api_directive.go

```diff
@@ -324,6 +324,20 @@ func (api *API) pushJobsTableKeys(ctx context.Context, jobs chan<- directiveJobT
 	// Get the diff between from/to directive.partitions.
 	partComp := newPartitionsComparer(fromD.TranslatePartitionsMap(), toPartitionsMap)
 
+	// Remove any partitions which are no longer assigned to this worker.
+	// TODO(tlt): currently, this is just removing the file lock on the
+	// resource; it's not actually removing the resource from the local
+	// computer. We should do that.
+	for tkey, partitions := range partComp.removed() {
+		qtid := tkey.QualifiedTableID()
+		for _, partition := range partitions {
+			resource := api.serverlessStorage.GetTableKeyResource(qtid, partition)
```

Comment:
serverlessStorage has `RemoveTableKeyResource` which unlocks the resource as it removes it

Reply:
(same for all the other types of resources of course)

Reply:
wait, I don't want to actually remove the resource though right? because another node will be taking control of it. I thought I just need to release the lock. Or am I misunderstanding this?

Reply:
you are misunderstanding (which is totally unrelated to my complete lack of doc comments), removal just removes the resource from management, it does not delete it. Deletion is handled by the controller, so there's no API in the storage manager to delete resources.

Reply:
changed it to use the `Remove*Resource` methods